GoogleSignIn/Tests/Unit/GIDEMMSupportTest.m

```diff
@@ -0,0 +1,210 @@
+  [GULSwizzler swizzleClass:[UIDevice class]
+                    selector:@selector(systemName)
+             isClassSelector:NO
+                   withBlock:^(id sender) { return kOldIOSName; }];
+
+  NSDictionary *originalParameters = @{
+    @"emm_support" : @"xyz",
+  };
+
+  NSDictionary *updatedEMMParameters =
+      [GIDEMMSupport updatedEMMParametersWithParameters:originalParameters];
+
+  NSDictionary *expectedParameters = @{
+    @"emm_support" : @"xyz",
+    @"device_os" : [NSString stringWithFormat:@"%@ %@",
+                    kNewIOSName, [UIDevice currentDevice].systemVersion]
+  };
+
+  XCTAssertEqualObjects(updatedEMMParameters, expectedParameters);
+
+
+  [GULSwizzler unswizzleClass:[UIDevice class]
+                     selector:@selector(systemName)
+              isClassSelector:NO];
+}
+
+- (void)testUpdateEMMParametersWithParameters_passcodInfo {
+  [GULSwizzler swizzleClass:[UIDevice class]
+                   selector:@selector(systemName)
+            isClassSelector:NO
+                  withBlock:^(id sender) { return kOldIOSName; }];
+
+  NSDictionary *originalParameters = @{
+    @"emm_support" : @"xyz",
+    @"device_os" : @"old one",
+    @"emm_passcode_info" : @"something",
+  };
+
+  NSDictionary *updatedEMMParameters =
+      [GIDEMMSupport updatedEMMParametersWithParameters:originalParameters];
+
+  NSDictionary *expectedParameters = @{
+    @"emm_support" : @"xyz",
+    @"device_os" : [NSString stringWithFormat:@"%@ %@",
+        kNewIOSName, [UIDevice currentDevice].systemVersion],
+    @"emm_passcode_info" : [GIDMDMPasscodeState passcodeState].info,
+  };
+
+  XCTAssertEqualObjects(updatedEMMParameters, expectedParameters);
+
+  [GULSwizzler unswizzleClass:[UIDevice class]
+                     selector:@selector(systemName)
+              isClassSelector:NO];
+}
+
+- (void)testHandleTokenFetchEMMError_errorIsEMM {
+  // Set expectations.
+  NSDictionary *errorJSON = @{ @"error" : @"EMM Specific Error" };
+  NSError *emmError = [NSError errorWithDomain:@"anydomain"
+                                          code:12345
+                                      userInfo:@{ OIDOAuthErrorResponseErrorKey : errorJSON }];
+  id mockEMMErrorHandler = OCMStrictClassMock([GIDEMMErrorHandler class]);
+  [[[mockEMMErrorHandler stub] andReturn:mockEMMErrorHandler] sharedInstance];
+  __block void (^savedCompletion)(void);
+  [[[mockEMMErrorHandler expect] andReturnValue:@YES]
+      handleErrorFromResponse:errorJSON completion:[OCMArg checkWithBlock:^(id arg) {
+    savedCompletion = arg;
+    return YES;
+  }]];
+
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Callback is called"];
+
+  [GIDEMMSupport handleTokenFetchEMMError:emmError completion:^(NSError *error) {
+    [expectation fulfill];
+    XCTAssertEqualObjects(error.domain, kGIDSignInErrorDomain);
+    XCTAssertEqual(error.code, kGIDSignInErrorCodeEMM);
+  }];
+
+  savedCompletion();
```

Comment:
Why did you decide to drop the inverted expectation from `testEMMError`?

Reply:
Added inverted expectation.